### PR TITLE
docs: AsciiDoc docs for publishing

### PR DESCRIPTION
**Pull request by trentm**
TODO:
- [x] complete the content from the README and trim those
- [x] Get help (from Brandon?) to review and setup publishing these to https://www.elastic.co/guide/en/ecs-logging/index.html

### CODE REVIEW

**Comment by trentm** [comment]
@bmorelli25 I would appreciate your help on hooking this into publishing and, of course, any content and structure feedback you have. I still have more content from the READMEs (generally I mark with an "XXX" where things are incomplete).

**Comment by bmorelli25** [comment]
You were onto something with your comment – all anchors do need to be unique. I've added a commit to this PR that fixes that problem.

To build these docs locally, make sure you have an up-to-date copy of `elastic/docs` and `elastic/ecs-logging`. If you don't export a `GIT_HOME` attribute, you'll need to do that too: `export GIT_HOME="/<fullPathTYourRepos>"` Then run:

```
$GIT_HOME/docs/build_docs --doc $GIT_HOME/ecs-logging-js/docs/index.asciidoc --resource=$GIT_HOME/ecs-logging/docs/ --chunk 1 --open
```

File: docs/morgan.asciidoc [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+----
+<1> Pass the ECS formatter to `morgan()`.
+
+See usage discussion and examples below.
+
+[float]
+[[morgan-setup-step-3]]
+==== Step 3: Configure Filebeat
+
+The best way to collect the logs once they are ECS-formatted is with https://www.elastic.co/beats/filebeat[Filebeat]:
```
Comment (bmorelli25) [inline docs/morgan.asciidoc:50]: Here, I think it makes more sense to link to the docs than the product page.

```suggestion
The best way to collect the logs once they are ECS-formatted is with {filebeat-ref}[Filebeat]:
```

File: docs/pino.asciidoc [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+----
+<1> This will https://getpino.io/#/docs/api?id=options[configure] Pino's `formatters`, `messageKey` and `timestamp` options.
+
+See usage discussion and examples below.
+
+[float]
+[[pino-setup-step-3]]
+==== Step 3: Configure Filebeat
+
+The best way to collect the logs once they are ECS-formatted is with https://www.elastic.co/beats/filebeat[Filebeat]:
```
Comment (bmorelli25) [inline docs/pino.asciidoc:45]:
```suggestion
The best way to collect the logs once they are ECS-formatted is with {filebeat-ref}[Filebeat]:
```

File: docs/winston.asciidoc [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+----
+<1> Pass the ECS formatter to winston here.
+
+See usage discussion and examples below.
+
+[float]
+[[winston-setup-step-3]]
+==== Step 3: Configure Filebeat
+
+The best way to collect the logs once they are ECS-formatted is with https://www.elastic.co/beats/filebeat[Filebeat]:
```
Comment (bmorelli25) [inline docs/winston.asciidoc:50]:
```suggestion
The best way to collect the logs once they are ECS-formatted is with {filebeat-ref}[Filebeat]:
```

**Review by bmorelli25 — APPROVED** [review]
Haven't tested any of the content, but the structure and big-picture looks good. I've opened https://github.com/elastic/docs/pull/2048 to add this to the doc build. This PR will need to be merged first. Once the doc build PR is merged, the webhook will kick-in and we'll get doc-checks on any subsequent PRs.

File: docs/morgan.asciidoc (hunk 1 above)
Comment (trentm) [reply docs/morgan.asciidoc:50]: Sounds good to me. FWIW, I copied this from https://github.com/elastic/ecs-logging-python/blob/master/docs/setup.asciidoc#install-filebeat

**Comment by trentm** [comment]
@bmorelli25 I'd appreciate a sanity content review from you if you have the time.

**Comment by gtback** [comment]
@elasticmachine run elasticsearch-ci/docs

I'm hoping with this comment it will test the docs build.

File: docs/intro.asciidoc [hunk 4]
```diff
@@ -0,0 +1,33 @@
+[[intro]]
+== Introduction
+
+beta::[]
+
+Node.js ECS loggers are formatter plugins for your favorite logging libraries.
+They make it easy to format your logs into ECS-compatible JSON. In combination
+with https://www.elastic.co/products/beats/filebeat[filebeat] you can send your
+logs directly to Elasticsearch and leverage https://www.elastic.co/guide/en/infrastructure/guide/current/logs-ui-overview.html[Kibana's Logs UI]
```
Comment (bmorelli25) [inline docs/intro.asciidoc:9]: These docs were deprecated in 7.4. Here's the updated link:
```suggestion
logs directly to Elasticsearch and leverage {observability-guide}/monitor-logs.html[Kibana's Logs app]
```

File: docs/intro.asciidoc [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+to inspect all logs in one single place.
+
+The Node.js ECS logging formatters log structured JSON and support serialization
+of Error objects, and HTTP Request and Response objects from Node.js core and
```
Comment (bmorelli25) [inline docs/intro.asciidoc:13]:
```suggestion
of Error objects and HTTP Request and Response objects from Node.js core and
```

File: README.md [hunk 6]
```diff
@@ -6,7 +6,8 @@
 
 This set of libraries allows you to transform your application logs to structured logs that comply with the [Elastic Common Schema (ECS)](https://www.elastic.co/guide/en/ecs/current/ecs-reference.html).
 In combination with [filebeat](https://www.elastic.co/products/beats/filebeat) you can send your logs directly to Elasticsearch and leverage [Kibana's Logs UI](https://www.elastic.co/guide/en/infrastructure/guide/current/logs-ui-overview.html) to inspect all logs in one single place.
```
Comment (bmorelli25) [inline README.md:8]: "Logs app" is the offical term

```suggestion
In combination with [filebeat](https://www.elastic.co/products/beats/filebeat) you can send your logs directly to Elasticsearch and leverage [Kibana's Logs app](https://www.elastic.co/guide/en/observability/current/monitor-logs.html) to inspect all logs in one single place.
```

File: loggers/morgan/README.md [hunk 7]
```diff
@@ -10,7 +10,9 @@ This Node.js package provides a formatter for the
 In combination with the [filebeat](https://www.elastic.co/products/beats/filebeat)
 shipper, you can send your logs directly to Elasticsearch and leverage
 [Kibana's Logs UI](https://www.elastic.co/guide/en/infrastructure/guide/current/logs-ui-overview.html)
```
Comment (bmorelli25) [inline loggers/morgan/README.md:12]:
```suggestion
[Kibana's Logs app](https://www.elastic.co/guide/en/observability/current/monitor-logs.html)
```

File: loggers/pino/README.md [hunk 8]
```diff
@@ -11,6 +11,8 @@ shipper, you can send your logs directly to Elasticsearch and leverage
 [Kibana's Logs UI](https://www.elastic.co/guide/en/infrastructure/guide/current/logs-ui-overview.html)
```
Comment (bmorelli25) [inline loggers/pino/README.md:11]:
```suggestion
[Kibana's Logs app](https://www.elastic.co/guide/en/observability/current/monitor-logs.html)
```

File: loggers/winston/README.md [hunk 9]
```diff
@@ -12,6 +12,8 @@ shipper, you can send your logs directly to Elasticsearch and leverage
 [Kibana's Logs UI](https://www.elastic.co/guide/en/infrastructure/guide/current/logs-ui-overview.html)
```
Comment (bmorelli25) [inline loggers/winston/README.md:12]:
```suggestion
[Kibana's Logs app](https://www.elastic.co/guide/en/observability/current/monitor-logs.html)
```

File: docs/morgan.asciidoc [hunk 10]
```diff
@@ -0,0 +1,197 @@
...
+
+
+[float]
+[[morgan-apm]]
+=== Integration with APM Tracing
+
+This ECS log formatter integrates with https://www.elastic.co/apm[Elastic APM] tracing.
+If your Node app is using the https://www.elastic.co/guide/en/apm/agent/nodejs/current/intro.html[Node.js Elastic APM Agent],
+then fields are added to log records that https://www.elastic.co/guide/en/ecs/current/ecs-tracing.html[identify an active trace] and the configured service name
+(https://www.elastic.co/guide/en/ecs/current/ecs-service.html#field-service-name["service.name"] and
```
Comment (bmorelli25) [inline docs/morgan.asciidoc:170]:
```suggestion
({ecs-ref}/ecs-service.html#field-service-name["service.name"] and
```

File: docs/morgan.asciidoc [hunk 11]
```diff
@@ -0,0 +1,197 @@
...
+https://www.elastic.co/guide/en/ecs/current/ecs-event.html#field-event-dataset["event.dataset"]).
```
Comment (bmorelli25) [inline docs/morgan.asciidoc:171]:
```suggestion
{ecs-ref}/ecs-event.html#field-event-dataset["event.dataset"]).
```

File: docs/morgan.asciidoc [hunk 12]
```diff
@@ -0,0 +1,197 @@
...
+----
+
+
+[float]
+[[morgan-apm]]
+=== Integration with APM Tracing
+
+This ECS log formatter integrates with https://www.elastic.co/apm[Elastic APM] tracing.
+If your Node app is using the https://www.elastic.co/guide/en/apm/agent/nodejs/current/intro.html[Node.js Elastic APM Agent],
+then fields are added to log records that https://www.elastic.co/guide/en/ecs/current/ecs-tracing.html[identify an active trace] and the configured service name
```
Comment (bmorelli25) [inline docs/morgan.asciidoc:169]:
```suggestion
then fields are added to log records that {ecs-ref}/ecs-tracing.html[identify an active trace] and the configured service name
```

File: docs/morgan.asciidoc [hunk 13]
```diff
@@ -0,0 +1,197 @@
...
+  ...
+----
+
+
+[float]
+[[morgan-apm]]
+=== Integration with APM Tracing
+
+This ECS log formatter integrates with https://www.elastic.co/apm[Elastic APM] tracing.
+If your Node app is using the https://www.elastic.co/guide/en/apm/agent/nodejs/current/intro.html[Node.js Elastic APM Agent],
```
Comment (bmorelli25) [inline docs/morgan.asciidoc:168]:
```suggestion
If your Node app is using the {apm-node-ref}/intro.html[Node.js Elastic APM Agent],
```

File: docs/pino.asciidoc [hunk 14]
```diff
@@ -0,0 +1,261 @@
...
+{"log.level":"info","@timestamp":"2021-01-19T22:51:12.142Z","ecs":{"version":"1.5.0"},"process":{"pid":82240},"host":{"hostname":"pink.local"},"message":"Hello world"}
+{"log.level":"warn","@timestamp":"2021-01-19T22:51:12.143Z","ecs":{"version":"1.5.0"},"process":{"pid":82240},"host":{"hostname":"pink.local"},"module":"foo","message":"From child"}
+----
+
+[float]
+[[pino-error-logging]]
+=== Error Logging
+
+By default, the formatter will convert an `err` field that is an Error instance
+to https://www.elastic.co/guide/en/ecs/current/ecs-error.html[ECS Error fields].
```
Comment (bmorelli25) [inline docs/pino.asciidoc:80]:
```suggestion
to {ecs-ref}/ecs-error.html[ECS Error fields].
```

File: docs/pino.asciidoc [hunk 15]
```diff
@@ -0,0 +1,261 @@
...
+
+server.listen(3000, () => {
+  log.info('listening at http://localhost:3000')
+}
+----
+<1> use `convertReqRes` option
+<2> log with `req` and/or `res` fields
+
+This will produce logs with request and response info using
+https://www.elastic.co/guide/en/ecs/current/ecs-http.html[ECS HTTP fields].
```
Comment (bmorelli25) [inline docs/pino.asciidoc:153]:
```suggestion
{ecs-ref}/ecs-http.html[ECS HTTP fields].
```

File: docs/pino.asciidoc [hunk 16]
```diff
@@ -0,0 +1,261 @@
...
+
+[float]
+[[pino-apm]]
+=== Integration with APM Tracing
+
+This ECS log formatter integrates with https://www.elastic.co/apm[Elastic APM].
+If your Node app is using the https://www.elastic.co/guide/en/apm/agent/nodejs/current/intro.html[Node.js Elastic APM Agent],
+then fields are added to log records that https://www.elastic.co/guide/en/ecs/current/ecs-tracing.html[identify an active trace] and the configured service name
+(https://www.elastic.co/guide/en/ecs/current/ecs-service.html#field-service-name["service.name"] and
+https://www.elastic.co/guide/en/ecs/current/ecs-event.html#field-event-dataset["event.dataset"]).
```
Comment (bmorelli25) [inline docs/pino.asciidoc:214]:
```suggestion
If your Node app is using the {apm-node-ref}/intro.html[Node.js Elastic APM Agent],
then fields are added to log records that {ecs-ref}ecs-tracing.html[identify an active trace] and the configured service name
({ecs-ref}/ecs-service.html#field-service-name["service.name"] and
{ecs-ref}/ecs-event.html#field-event-dataset["event.dataset"]).
```

File: docs/winston.asciidoc [hunk 17]
```diff
@@ -0,0 +1,257 @@
...
+timestamp is automatically generated by the formatter, so you don't need to add
+the https://github.com/winstonjs/logform#timestamp[timestamp] formatter.
+
+
+[float]
+[[winston-error-logging]]
+=== Error logging
+
+By default, the formatter will convert an `err` meta field that is an Error instance
+to https://www.elastic.co/guide/en/ecs/current/ecs-error.html[ECS Error fields].
```
Comment (bmorelli25) [inline docs/winston.asciidoc:95]:
```suggestion
to {ecs-ref}/ecs-error.html[ECS Error fields].
```

File: docs/winston.asciidoc [hunk 18]
```diff
@@ -0,0 +1,257 @@
...
+  res.setHeader('Foo', 'Bar')
+  res.end('ok')
+  logger.info('handled request', { req, res }) <2>
+}
+----
+<1> use `convertReqRes` option
+<2> log `req` and/or `res` meta fields
+
+This will produce logs with request and response info using
+https://www.elastic.co/guide/en/ecs/current/ecs-http.html[ECS HTTP fields].
```
Comment (bmorelli25) [inline docs/winston.asciidoc:183]:
```suggestion
{ecs-ref}/ecs-http.html[ECS HTTP fields].
```

File: docs/winston.asciidoc [hunk 19]
```diff
@@ -0,0 +1,257 @@
...
+
+[float]
+[[winston-apm]]
+=== Integration with APM Tracing
+
+This ECS log formatter integrates with https://www.elastic.co/apm[Elastic APM].
+If your Node app is using the https://www.elastic.co/guide/en/apm/agent/nodejs/current/intro.html[Node.js Elastic APM Agent],
+then fields are added to log records that https://www.elastic.co/guide/en/ecs/current/ecs-tracing.html[identify an active trace] and the configured service name
+(https://www.elastic.co/guide/en/ecs/current/ecs-service.html#field-service-name["service.name"] and
+https://www.elastic.co/guide/en/ecs/current/ecs-event.html#field-event-dataset["event.dataset"]).
```
Comment (bmorelli25) [inline docs/winston.asciidoc:232]:
```suggestion
If your Node app is using the {apm-node-ref}/intro.html[Node.js Elastic APM Agent],
then fields are added to log records that {ecs-ref}/ecs-tracing.html[identify an active trace] and the configured service name
({ecs-ref}/ecs-service.html#field-service-name["service.name"] and
{ecs-ref}/ecs-event.html#field-event-dataset["event.dataset"]).
```

**Review by bmorelli25 — COMMENTED** [review]
Most of these are updates to attributes that enable internal link checking.

**Review by bmorelli25 — APPROVED** [review]
Tested locally one final time–everything looks good!

**Pull request merged**